### PR TITLE
 add opt-out flag for quote adjustments in getOrderToSign

### PR DESCRIPTION
**Pull request by carlos-cow**
## Description

  Adds an `applyQuoteAdjustments` flag to `getOrderToSign` so callers can opt out of slippage/partner-fee amount adjustments when reusing raw quote values. Includes unit tests that cover both SELL and BUY flows with
  the flag disabled.

## Context

  Some downstream consumers need to preserve the original quote amounts (e.g., when custom adjustments are applied elsewhere). Without a way to bypass the internal `getQuoteAmountsAndCosts` call, they must duplicate logic or patch results manually. This flag keeps the existing default behavior while enabling those advanced use cases.

## Out of Scope
N/A

## Testing Instructions

```bash
pnpm --filter @cowprotocol/sdk-trading test -- --runTestsByPath src/getOrderToSign.test.ts
```

## Summary by CodeRabbit

* **New Features**
  * Option to disable quote adjustments when preparing orders for signing.

* **Updates**
  * Flash loan fee input now uses basis-points and applies Aave-style rounding (ceil on remainder), yielding more accurate fee and signed amounts.
  * Signing payload now includes the order-to-sign flag to opt out of quote adjustments.

* **Tests**
  * Expanded tests covering flash loan fee edge cases, rounding behaviors, and quote-adjustment scenarios.

### CODE REVIEW

File: packages/trading/src/types.ts [hunk 1]
```diff
@@ -282,4 +282,10 @@ export interface PostTradeAdditionalParams {
    */
   signingScheme?: SigningScheme
   customEIP1271Signature?: (orderToSign: UnsignedOrder, signer: AbstractSigner<Provider>) => Promise<string>
+
+  /**
+   * By default, the quote adjustments are applied to the order.
```
Comment (anxolin) [inline packages/trading/src/types.ts]: For me the description of this was not enough to understand. 

Could you indicate that this "quote adjustments" to applying to the sold/bought amounts the gas costs, slippage, partner fees, etc.

Setting it to `false` will post the order with the un-modified amounts indicated in the parameters. Note that in this case it is assumed that these concepts were already applied by the caller of the function.
